### PR TITLE
Docupdates

### PR DESCRIPTION
**Pull request by OkonSamuel**
Switches doc generation to use Documenter

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -4,6 +4,7 @@ version = "0.6.1"
 
 [deps]
 Compat = "34da2185-b29b-5c13-b0c7-acf172513d20"
+Conda = "8f4d0f93-b110-5947-807f-2305c1781a2d"
```
Comment (cstjean) [inline Project.toml:7]: I'm a bit puzzled by this. Didn't we already depend on Conda?

File: .gitignore [hunk 2]
```diff
@@ -1,3 +1,4 @@
 /.DS_Store
 /site
+docs/biuld/
```
Comment (cstjean) [inline .gitignore:3]:
```suggestion
docs/build/
```

Comment (cstjean) [reply .gitignore:3]: I think because of this typo you committed a lot of files that shouldn't have been committed.

File: src/Skcore.jl [hunk 3]
```diff
@@ -115,20 +120,12 @@ symbols_in(::Any) = Set()
...
-   version = VersionNumber(version_)
-       
+   global import_already_warned
+   mod = PyCall.pyimport_conda("sklearn", "scikit-learn") 
+   version = vparse(mod.__version__)          
+   
    min_version = v"0.18.0"
+
    if version < min_version && !import_already_warned
         @warn("Your Python's scikit-learn has version $version. We recommend updating to $min_version or higher for best compatibility with ScikitLearn.jl.")
```
Comment (cstjean) [inline src/Skcore.jl:130]: I wonder if we shouldn't be using `@warn(..., once=true)`, but that can go in another PR.

**Comment by cstjean** [comment]
I understand that it's a bit of a hassle sometimes, but in general, it's a lot easier to review and merge quickly small PRs. Ideally this would have been 3 PRs,

 - Using `VersionParse`
 - Update for latest `DataFrames`
 - Documenter change

I don't think it's particularly critical here, so let's keep this great PR as is! 🎉

**Comment by cstjean** [comment]
Is there any way to preview the new docs?

**Comment by OkonSamuel** [comment]
> Is there any way to preview the new docs?

You have to build my branch locally

File: Project.toml (hunk 1 above)
Comment (OkonSamuel) [reply Project.toml:7]: I'll switch that extras

Comment (cstjean) [reply Project.toml]: I just want to understand what's going on, it's been a while. When a new user `]add ScikitLearn`, it should generally work. From some investigation, it [seems](https://github.com/ValdarT/ScikitLearn.jl/blob/5bbb33fc58e9e7b5199be962d47cec2b0b987bb3/REQUIRE) that `Conda` was in `REQUIRE`. I guess when we created Project.toml, we forgot to put it in there. So I think this PR is correct. `Conda` should be in `Project.toml`. No?

File: test/runtests.jl [hunk 4]
```diff
@@ -1,15 +1,15 @@
...
 
+import PyCall
+
 #use non-mkl versions of python packages (to allow MacOS tests pass)
 @static if Sys.isapple()
-    using Conda
-    Conda.add("nomkl")
-    Conda.rm("mkl")
+    PyCall.Conda.add("nomkl")
+    PyCall.Conda.rm("mkl")
```
Comment (cstjean) [inline test/runtests.jl:9]: Yeah, this looks wrong, it just reinforces that we should have `Conda` in `Project.toml` as you did it first, so that we can `import Conda` normally.

File: Project.toml (hunk 1 above)
Comment (OkonSamuel) [reply Project.toml:7]: Installing via PyCall seemed better though. I hate to have to force the user to user to use Conda by default. They may already have a `sklearn` installed on a different python and may want to use it.

Comment (cstjean) [reply Project.toml:7]: `@sk_import` calls `import_sklearn`, which calls `PyCall.pyimport_conda`, which first tries to import the module

```julia
 try
        pyimport(modulename)
```

If the user already has a working `scikit-learn` installed (accessible via PyCall), it should be used, and `Conda` won't install anything. I do believe that's for the best. A lot of `ScikitLearn.jl` users are relatively new to julia, and making their lives easier with good defaults is important.

File: docs/src/man/quickstart.md [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+ "virginica"
+ "virginica"
+ "virginica"
+
+```
+
+Next, we load the LogisticRegression model from scikit-learn's [library](models.md). This would require scikit-learn. See [Installation](models.md#Installation-1)
+
+
+```jldoctest quickstart
```
Comment (cstjean) [inline docs/src/man/quickstart.md:87]: The `jldoctest`s are a nice touch.

Comment (OkonSamuel) [reply docs/src/man/quickstart.md:87]: yes.
The only `jldoctest` failing now is as a result of changes in `GaussianProcesses` API.
I have opened a PR there which has been merged. Am still waiting for a new release to be tagged.

**Comment by OkonSamuel** [comment]
@cstjean  any further addition/substractions before i merge this ?

File: test/runtests.jl (hunk 4 above)
Comment (cstjean) [reply test/runtests.jl:9]: Could you please readd Conda to Project.toml and change the calls here to `Conda.add/rm`? Conda is installed anyway via PyCall, and it's just cleaner to directly call it/require it.
